src/main/java/com/commercetools/service/paypalPlus/impl/PaypalPlusPaymentServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.commercetools.service.paypalPlus.impl;
+
+import com.commercetools.exception.PaypalPlusServiceException;
+import com.commercetools.service.paypalPlus.PaypalPlusPaymentService;
+import com.paypal.api.payments.Payment;
+import com.paypal.base.rest.APIContext;
+import com.paypal.base.rest.PayPalRESTException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * <b>Note:</b> since {@link APIContext} caches <i>PayPal-Request-Id</i> one should not reuse the same instance of
+ * {@link PaypalPlusPaymentServiceImpl} for multiple Paypal requests.
+ */
+@Service
+@Scope("prototype")
+public class PaypalPlusPaymentServiceImpl extends BasePaypalPlusService implements PaypalPlusPaymentService {
```

Comment:
I see you already make an abstract class here, but I guess for now it's quite useless since only `PaypalPlusPaymentServiceImpl` extend it. Maybe next time just wait until we have really a need to abstract this class, e.g. when we will implement `PaypalRefundService`, so we know exactly what to put to the `BasePaypalPlusService`?